include/nccl_ofi_param.h

```diff
@@ -138,6 +138,17 @@ OFI_NCCL_PARAM_INT(cq_read_count, "CQ_READ_COUNT", 4);
  */
 OFI_NCCL_PARAM_STR(protocol, "PROTOCOL", NULL);
 
+/*
+ * Override the platform default for domain allocation, with
+ * respect to the device or endpoint.
+ *
+ * -1 (unset default): use the platform-specific configuration.
+ * 0: Allocate one domain per device
+ * 1: Allocate one domain per endpoint
+ */
+
+OFI_NCCL_PARAM_INT(domain_per_endpoint, "DOMAIN_PER_ENDPOINT", -1);
```

Comment:
This option isn't going to make sense to a user.  It's really saying "Create a Libfabric domain per thread" vs. "Create a Libfabric domain per process".

Reply:
Maybe s/domain_per_endpoint/create_libfabric_domain_per_thread/, plus a better comment?

Or do you have a better suggestion?